### DISCUSSION TITLE
cascade default

### DISCUSSION THREAD

**Discussion by ZehuaZhang** [prisma/prisma#21929] [Q&A] [answered]
### Question

What's the default value of cascade?

onDelete: SetNull
onUpdate: Cascade

**Comment by janpio** [comment, marked as answer]
From our documentation: https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-actions#referential-action-defaults
<img width="525" alt="image" src="https://github.com/prisma/prisma/assets/183673/1beb30a0-913b-4c96-a5a2-11d141c030a8">

**Reply by ZehuaZhang** [reply]
I actually read that, and I want to confirm here.
Could we update the doc, so that is shows in the API , rather than here
![image](https://github.com/prisma/prisma/assets/15708284/ce3eb8c2-bc03-4376-a901-aac361ad4fd9)

And to confirm, onUpdate cascade is by default, but onDelete I have to set to cascade manually

**Reply by janpio** [reply]
> I actually read that, and I want to confirm here.

I would expect the docs to be correct, yes.

> Could we update the doc, so that is shows in the API , rather than here

What do you mean? Should be it be put in the old upgrade path documentation? If so, where exactly?

> And to confirm, onUpdate cascade is by default, but onDelete I have to set to cascade manually

Yes, that is what the docs say. If you want to be 100% sure, just define both explicitly.

**Reply by ZehuaZhang** [reply]
Thank you. Could we update it in [Prisma API](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#arguments-6) and [Prisma Concept](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-actions#cascade)

**Reply by janpio** [reply]
What exactly do you want to update there? Does it give misleading information that I am missing?

**Reply by ZehuaZhang** [reply]
![image](https://user-images.githubusercontent.com/183673/282567840-1beb30a0-913b-4c96-a5a2-11d141c030a8.png)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
